opencti-platform/opencti-front/src/private/components/common/lists/FilterDate.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { Dispatch, FunctionComponent, KeyboardEvent } from 'react';
+import TextField from '@mui/material/TextField';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { useFormatter } from '../../../../components/i18n';
+
+interface FilterDateProps {
+  defaultHandleAddFilter: (k: string, id: string, value: Record<string, unknown> | string, event?: React.KeyboardEvent) => void
+  filterKey: string
+  inputValues: Record<string, string | Date>
+  setInputValues: Dispatch<Record<string, string | Date>>
+  variant: string
+}
+
+const FilterDate: FunctionComponent<FilterDateProps> = ({
+  defaultHandleAddFilter,
+  filterKey,
+  inputValues,
+  setInputValues,
+  variant,
+}) => {
+  const { t, nsd } = useFormatter();
+
+  const handleChangeDate = (date: Date) => {
+    setInputValues({ ...inputValues, [filterKey]: date });
+  };
+
+  const handleAcceptDate = (date: Date) => {
+    if (date && date.toISOString()) {
+      defaultHandleAddFilter(filterKey, date.toISOString(), nsd(date));
+    }
+  };
+
+  const handleValidateDate = (event: KeyboardEvent<HTMLElement>) => {
```

Comment:
Why does this have a return value ? is this really necessary

Reply:
It's not necessary, I will remove it.